File: mns-reactor/src/main/java/cz/fb/manaus/reactor/categorizer/AbstractProposerCategorizer.java
package cz.fb.manaus.reactor.categorizer;

import com.google.common.base.MoreObjects;
import cz.fb.manaus.core.category.categorizer.SettledBetCategorizer;
import cz.fb.manaus.core.model.BetAction;
import cz.fb.manaus.core.model.SettledBet;
import cz.fb.manaus.core.model.Side;
import cz.fb.manaus.reactor.betting.action.BetUtils;
import org.springframework.beans.factory.annotation.Autowired;

import java.util.List;
import java.util.Map;

import static com.google.common.base.Joiner.on;
import static com.google.common.base.Strings.emptyToNull;

public abstract class AbstractProposerCategorizer implements SettledBetCategorizer {
    @Autowired
    private BetUtils betUtils;

    protected List<String> getProposers(SettledBet settledBet) {
        Map<String, String> properties = settledBet.getBetAction().getProperties();
        String rawProposers = MoreObjects.firstNonNull(emptyToNull(properties.get(BetAction.PROPOSER_PROP)), "none");

        return betUtils.parseProposers(rawProposers);
    }

    protected String getSideAware(String prefix, Side side, String category) {
        return prefix + on('.').join(side.name().toLowerCase(), category);
    }
}

File: mns-reactor/src/main/java/cz/fb/manaus/reactor/categorizer/AbstractProposerCategorizer.kt
package cz.fb.manaus.reactor.categorizer

import com.google.common.base.Joiner.on
import com.google.common.base.MoreObjects
import com.google.common.base.Strings.emptyToNull
import cz.fb.manaus.core.category.categorizer.RealizedBetCategorizer
import cz.fb.manaus.core.model.BetAction
import cz.fb.manaus.core.model.RealizedBet
import cz.fb.manaus.core.model.Side
import cz.fb.manaus.reactor.betting.action.BetUtils
import org.springframework.beans.factory.annotation.Autowired

abstract class AbstractProposerCategorizer : RealizedBetCategorizer {
    @Autowired
    private lateinit var betUtils: BetUtils

    protected fun getProposers(realizedBet: RealizedBet): List<String> {
        val properties = realizedBet.betAction.properties
        val rawProposers = MoreObjects.firstNonNull(emptyToNull(properties[BetAction.PROPOSER_PROP]), "none")
        return betUtils.parseProposers(rawProposers)
    }

    protected fun getSideAware(prefix: String, side: Side, category: String): String {
        return prefix + on('.').join(side.name.toLowerCase(), category)
    }
}
